sdk/agentserver/azure-ai-agentserver-langgraph/samples/tool_client_example/mcp_with_tool_executor.py

```diff
@@ -0,0 +1,113 @@
+# Copyright (c) Microsoft. All rights reserved.
+"""Enhanced MCP example using ToolClient with AzureAIToolClient.
+
+This sample demonstrates how to use the ToolClient to integrate Azure AI
+Tool Client (which supports both MCP tools and Azure AI Tools API) with
+LangGraph's create_react_agent.
+"""
+
+import asyncio
+import os
+
+from dotenv import load_dotenv
+from importlib.metadata import version
+from langchain_openai import AzureChatOpenAI
+from langgraph.checkpoint.memory import MemorySaver
+
+from azure.ai.agentserver.core.client.tools.aio import AzureAIToolClient
+from azure.ai.agentserver.langgraph import ToolClient, from_langgraph
+from azure.identity.aio import DefaultAzureCredential
+
+load_dotenv()
+
+
+def create_agent(model, tools, checkpointer):
+    """Create a LangGraph agent based on the version."""
+    # for different langgraph versions
+    langgraph_version = version("langgraph")
+    if langgraph_version < "1.0.0":
```

Comment:
Don't do a string comparison for version numbers. 

In this case we should try to import the method and fall back to the older version rather than doing a version comparison. If we really want to show both in the example. We could also just add a comment on how to do it or just rely on langgraph < 2 customers to know what to do.